### ISSUE TITLE
React-Redux: mistaken spread on state.flags

### ISSUE THREAD

**Issue by jstclair** [tdeekens/flopflip#625]
**Describe the bug**
When using the `@flopflip/react-redux` package after the last release, when a flag is updated, only the updated flag is present in Redux state, not the merged { prevState, newFlag }. In our case, this is in combination with the `launchdarkly-adapter`. It's possible this was introduced with the fix for #613 

**To Reproduce**
Configure @flopflip with launchDarkly adapter and react-redux.

1. Given turn a flag on or off in LaunchDarkly
2. Then the redux flag state only contains the currently-changed flag; all the others have been removed.

**Expected behavior**
Should have the new flag + the previous flags.

**Screenshots**
![image](https://user-images.githubusercontent.com/365694/54278990-33ea9a00-4594-11e9-8f88-67886b5be532.png)

**Additional context**
We have tracked this down to a recent change in `flopflip/packages/react-redux/modules/ducks/flags/flags.ts`:

The current code is in the reducer is:
```
switch (action.type) {
    case UPDATE_FLAGS:
      return {
        ...state.flags,
        ...action.payload.flags,
      };

    default:
      return state;
  }
```
However, it's the state itself that contains the flags; since state.flags is undefined, the object spread produces just the latest flag as the new state. The correct code should be:

```
case UPDATE_FLAGS:
  return { ...state, ...action.payload.flags };
```

**Comment by tdeekens** [comment]
Snap. Thanks will fix later tonite.

**Comment by tdeekens** [comment]
Thanks again for the very detailed bug report. Really easy to follow along. I took a shot at fixing the issue in #626. I published a canary release with the version `8.0.2-next.9+e10be08`. You can install that from npm as normal (it's on the next dist-tag) by changing to that version in your `package.json`.

Let me know if that works. I don't use the redux version so I don't half a "real world" app with it to test with. However, the tests in the library were adjusted.

**Comment by jstclair** [comment]
Great! thanks - i'm on my way out but will check it from home this evening.

**Comment by jstclair** [comment]
Just confirmed that fixes it - thank you!

**Comment by tdeekens** [comment]
Thanks for the quick feedback. Will release tomorrow morning and ping you here again :)

**Comment by tdeekens** [comment]
Released as `@flopflip/react-redux@8.0.2`. Thanks again.
